include/seqan3/core/type_list/traits.hpp

```diff
@@ -35,6 +35,19 @@ constexpr ptrdiff_t find()
     return ((SEQAN3_IS_SAME(query_t, pack_t) ? false : ++i) && ...) ? -1 : i;
 }
 
+/*!\brief Implementation for seqan3::pack_traits::find_if.
+ * \tparam pred_t   The predicate that is being evaluated.
+ * \tparam pack_t   The type pack.
+ * \returns The position of the first type `t` in `pack_t` for whom `pred_t<t>::value` is true.
+ * \ingroup type_list
+ */
+template <template <typename> typename pred_t, typename ...pack_t>
+constexpr ptrdiff_t find_if()
+{
+    ptrdiff_t i = 0;
+    return ((pred_t<pack_t>::value ? false : ++i) && ...) ? -1 : i;
```

Comment:
I'm curious as to how this first ternary operator actually works!

Reply:
I'll explain it to you on monday!